SwiftCharts/ChartCoordsSpace.swift

```diff
@@ -118,7 +174,18 @@ public class ChartCoordsSpace {
                 layer.rect.width + chartSettings.spacingBetweenAxesY
             }, generator: generator)
     }
-    
+
+    /**
+     Uses a generator to make axis layers from axis models. This method is used for all axes.
+
+     - parameter axisModels:    The models used to generate the axis layers
+     - parameter offset:        The offset in points for the generated layers
+     - parameter pointsCreator: TODO: not sure exactly what this effectively does
```

Comment:
Yes, it takes the offset of the respective axis (this is its position relative to the start value -the start value depends if it's x/y/low/high, you can see what is being passed- along the dimension *orthogonal* to the axis) and returns p1 and p2 for that axis... `dimIncr` is the function that calculates the offset for the next axis if existent (when there is only one axis, this function is practically not used).

This method is just the result of refactoring the common code of all axes, since they can be vertical or horizontal and depending where the axis is, the offset for the next axis goes in a different direction I had to pass these functions to help with that.

My wording "multi-axes space" is maybe not the best, as this is also used for single axis, what I meant is basically across the dimension orthogonal to axis (which is the dimension along we "move" with `reduce` to generate multiple axes if existent), in the current group of axes (x low, y low, x high or y high).

One additional thing worth noting, is that the order of the indices of multiple axes, depends on the axes group, for example in y low the axes go from the furthest (index 0) to nearest (index n), while y low they go from nearest (index 0) to furthest (index n). This is why y low starts with an offset of `chartSettings.leading` which is the external margin or the chart, while x low starts with an offset of `chartFrame.origin.y + chartFrame.height` which is the lower edge of the chart *inner* frame. This is admittedly not intuitive, but it was quicker to implement! It could be improved at some point (index always increases inside->out) or could at least benefit from another graphic, which shows the index numbers ;)